python/lsst/pipe/tasks/healSparseMappingProperties.py

```diff
@@ -0,0 +1,490 @@
+            Array of approximate psf values matched to ra/dec.
+        """
+        raise NotImplementedError("All property maps must implement compute()")
+
+    def post_process(self, total_weights, total_inputs):
+        """Perform post-processing on values.
+
+        Parameters
+                ----------
+        total_weights : `np.ndarray`
+            Total accumulated weights, for each value index.
+        total_inputs : `np.ndarray`
+            Total number of inputs, for each value index.
+        """
+        # Override of this method is not required.
+        pass
+
+
+@register_property_map("exposure_time")
+class ExposureTimePropertyMap(BasePropertyMap):
+    """Exposure time property map."""
+
+    def compute(self, row, ra, dec, scalings, psf_array=None):
+        return row.getVisitInfo().getExposureTime()
+
+
+@register_property_map("psf_size")
+class PsfSizePropertyMap(BasePropertyMap):
+    """PSF size property map."""
+    requires_psf = True
+
+    def compute(self, row, ra, dec, scalings, psf_array=None):
+        return psf_array["psf_size"]
+
+
+@register_property_map("psf_e1")
+class PsfE1PropertyMap(BasePropertyMap):
+    """PSF shape e1 property map."""
+    requires_psf = True
+
+    def compute(self, row, ra, dec, scalings, psf_array=None):
+        return psf_array["psf_e1"]
+
+
+@register_property_map("psf_e2")
+class PsfE2PropertyMap(BasePropertyMap):
+    """PSF shape e2 property map."""
+    requires_psf = True
+
+    def compute(self, row, ra, dec, scalings, psf_array=None):
+        return psf_array["psf_e2"]
+
+
+@register_property_map("n_exposure")
+class NExposurePropertyMap(BasePropertyMap):
+    """Number of exposures property map."""
+    dtype = np.int32
+
+    def compute(self, row, ra, dec, scalings, psf_array=None):
+        return 1
+
+
+class PsfMaglimPropertyMapConfig(BasePropertyMapConfig):
+    """Configuration for the PsfMaglim property map.
+    """
+    maglim_nsigma = pexConfig.Field(dtype=float, default=5.0,
+                                    doc="Number of sigma to compute magnitude limit.")
+
+    def validate(self):
+        super().validate()
+        if self.do_min or self.do_max or self.do_mean or self.do_sum:
+            raise ValueError("Can only use do_weighted_mean with PsfMaglimPropertyMap")
+
+
+@register_property_map("psf_maglim")
+class PsfMaglimPropertyMap(BasePropertyMap):
+    """PSF magnitude limit property map."""
+    requires_psf = True
+
+    ConfigClass = PsfMaglimPropertyMapConfig
+
+    def compute(self, row, ra, dec, scalings, psf_array=None):
+        # Our values are the weighted mean of the psf area
+        return psf_array["psf_area"]
+
+    def post_process(self, total_weights, total_inputs):
+        psf_area = self.weighted_mean_values.copy()
+        maglim = (self.zeropoint
+                  - 2.5*np.log10(1./np.sqrt(total_weights))
+                  - 2.5*np.log10(self.config.maglim_nsigma*np.sqrt(psf_area)))
```

Comment:
```suggestion
                  maglim = (self.zeropoint - 2.5*np.log10(self.config.maglim_nsigma*np.sqrt(psf_area/total_weights)))
```
The intention being keeping the calls to `log10` to a minimum (I haven't paid attention to line length in my suggestion)